libraries/stdlib/jdk7/src/kotlin/io/PathReadWrite.kt

```diff
@@ -0,0 +1,373 @@
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Path.reader(charset: Charset = Charsets.UTF_8): InputStreamReader {
+    return inputStream().reader(charset)
+}
+
+/**
+ * Returns a new [InputStreamReader] for reading the content of this file.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Path.reader(vararg options: OpenOption, charset: Charset = Charsets.UTF_8): InputStreamReader {
+    return inputStream(*options).reader(charset)
+}
+
+/**
+ * Returns a new [BufferedReader] for reading the content of this file.
+ *
+ * @param charset character set to use.
+ * @param bufferSize necessary size of the buffer.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Path.bufferedReader(charset: Charset = Charsets.UTF_8, bufferSize: Int = DEFAULT_BUFFER_SIZE): BufferedReader {
+    return reader(charset).buffered(bufferSize)
+}
+
+/**
+ * Returns a new [BufferedReader] for reading the content of this file.
+ *
+ * @param options options to determine how the file is opened
+ * @param charset character set to use.
+ * @param bufferSize necessary size of the buffer.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Path.bufferedReader(
+    vararg options: OpenOption,
+    charset: Charset = Charsets.UTF_8,
+    bufferSize: Int = DEFAULT_BUFFER_SIZE
+): BufferedReader {
+    return reader(*options, charset = charset).buffered(bufferSize)
+}
+
+/**
+ * Returns a new [OutputStreamWriter] for writing the content of this file.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Path.writer(charset: Charset = Charsets.UTF_8): OutputStreamWriter {
+    return outputStream().writer(charset)
+}
+
+/**
+ * Returns a new [OutputStreamWriter] for writing the content of this file.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Path.writer(vararg options: OpenOption, charset: Charset = Charsets.UTF_8): OutputStreamWriter {
+    return outputStream(*options).writer(charset)
+}
+
+
+/**
+ * Returns a new [BufferedWriter] for writing the content of this file.
+ *
+ * @param charset character set to use.
+ * @param bufferSize necessary size of the buffer.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Path.bufferedWriter(charset: Charset = Charsets.UTF_8, bufferSize: Int = DEFAULT_BUFFER_SIZE): BufferedWriter {
+    return writer(charset).buffered(bufferSize)
+}
+
+/**
+ * Returns a new [BufferedWriter] for writing the content of this file.
+ *
+ * @param options options to determine how the file is opened
+ * @param charset character set to use.
+ * @param bufferSize necessary size of the buffer.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+@kotlin.internal.InlineOnly
+public inline fun Path.bufferedWriter(
+    vararg options: OpenOption,
```

Comment:
Yes, i went with this order because `charset` has a default value.

With the varargs first, you have to pass `charset` as a keyword argument, but you can omit it to use the defautl value.

With varargs last, you would have to specify an explicit value of `charset` in order to pass any vararg arguments.

I kept the overloads without vararg options so that you can pass charset without varargs.

Another possible design would be to instead have three overloads of each function, one with just a charset and default value, one with just varargs, and one with charset (with no default) followed by varargs. That would be allow you to pass both a charset and varargs without keyword arguments.

Reply:
I went ahead and moved the varargs to the last position and implemented all the extra overloads. The overloads are `@InlineOnly`, so it seems fine to add a few. 

I can always revert the change.